### ISSUE TITLE
Cargo test doesn't work with goldie

### ISSUE THREAD

**Issue by ivan-gusiev** [rossmacarthur/goldie#1]
*Step 1.* Create a new empty directory. Run `cargo init`.
*Step 2.* Run `cargo add --dev goldie`
*Step 3.* Run `cargo test`

This will result in the following error:
```
error: cannot find derive macro `Deserialize` in this scope
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/goldie-0.4.2/src/lib.rs:294:22
    |
294 |             #[derive(Deserialize)]
    |                      ^^^^^^^^^^^
    |
note: `Deserialize` is imported here, but it is only a trait, without a derive macro
   --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/goldie-0.4.2/src/lib.rs:64:13
    |
64  | use serde::{Deserialize, Serialize};
    |             ^^^^^^^^^^^

error[E0277]: the trait bound `Manifest: Deserialize<'_>` is not satisfied
    --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/goldie-0.4.2/src/lib.rs:305:38
     |
305  |             let manifest: Manifest = serde_json::from_slice(&output.stdout).unwrap();
     |                                      ^^^^^^^^^^^^^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `Manifest`
     |
     = help: the following other types implement trait `Deserialize<'de>`:
               &'a Path
               &'a [u8]
               &'a str
               ()
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
               (T0, T1, T2, T3, T4)
             and 127 others
note: required by a bound in `from_slice`
    --> /home/username/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_json-1.0.91/src/de.rs:2561:8
     |
2561 |     T: de::Deserialize<'a>,
     |        ^^^^^^^^^^^^^^^^^^^ required by this bound in `from_slice`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `goldie` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
```

Rust/cargo versions:
```
username@DESKTOP-AADV0LD:~/repos/goldie-test-repo$ cargo --version
cargo 1.65.0 (4bc8f24d3 2022-10-20)
username@DESKTOP-AADV0LD:~/repos/goldie-test-repo$ rustc --version
rustc 1.65.0 (897e37553 2022-11-02)
```

**Comment by ivan-gusiev** [comment]
Lock file contents:

```
# This file is automatically @generated by Cargo.
# It is not intended for manual editing.
version = 3

[[package]]
name = "anyhow"
version = "1.0.68"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "2cb2f989d18dd141ab8ae82f64d1a8cdd37e0840f73a406896cf5e99502fab61"

[[package]]
name = "ctor"
version = "0.1.26"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "6d2301688392eb071b0bf1a37be05c469d3cc4dbbd95df672fe28ab021e6a096"
dependencies = [
 "quote",
 "syn",
]

[[package]]
name = "diff"
version = "0.1.13"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "56254986775e3233ffa9c4d7d3faaf6d36a2c09d30b20687e9f88bc8bafc16c8"

[[package]]
name = "goldie"
version = "0.4.2"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "e5fae5db05aa608b6882a7c299ef8171f8665e0264d62fd6f0b12f6cf636450b"
dependencies = [
 "anyhow",
 "once_cell",
 "pretty_assertions",
 "serde",
 "serde_json",
 "upon",
 "yansi",
]

[[package]]
name = "itoa"
version = "1.0.5"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "fad582f4b9e86b6caa621cabeb0963332d92eea04729ab12892c2533951e6440"

[[package]]
name = "megacat"
version = "0.1.0"
dependencies = [
 "goldie",
]

[[package]]
name = "once_cell"
version = "1.17.0"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "6f61fba1741ea2b3d6a1e3178721804bb716a68a6aeba1149b5d52e3d464ea66"

[[package]]
name = "output_vt100"
version = "0.1.3"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "628223faebab4e3e40667ee0b2336d34a5b960ff60ea743ddfdbcf7770bcfb66"
dependencies = [
 "winapi",
]

[[package]]
name = "pretty_assertions"
version = "1.3.0"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "a25e9bcb20aa780fd0bb16b72403a9064d6b3f22f026946029acb941a50af755"
dependencies = [
 "ctor",
 "diff",
 "output_vt100",
 "yansi",
]

[[package]]
name = "proc-macro2"
version = "1.0.49"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "57a8eca9f9c4ffde41714334dee777596264c7825420f521abc92b5b5deb63a5"
dependencies = [
 "unicode-ident",
]

[[package]]
name = "quote"
version = "1.0.23"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "8856d8364d252a14d474036ea1358d63c9e6965c8e5c1885c18f73d70bff9c7b"
dependencies = [
 "proc-macro2",
]

[[package]]
name = "ryu"
version = "1.0.12"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "7b4b9743ed687d4b4bcedf9ff5eaa7398495ae14e61cba0a295704edbc7decde"

[[package]]
name = "serde"
version = "1.0.152"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "bb7d1f0d3021d347a83e556fc4683dea2ea09d87bccdf88ff5c12545d89d5efb"

[[package]]
name = "serde_json"
version = "1.0.91"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "877c235533714907a8c2464236f5c4b2a17262ef1bd71f38f35ea592c8da6883"
dependencies = [
 "itoa",
 "ryu",
 "serde",
]

[[package]]
name = "syn"
version = "1.0.107"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "1f4064b5b16e03ae50984a5a8ed5d4f8803e6bc1fd170a3cda91a1be4b18e3f5"
dependencies = [
 "proc-macro2",
 "quote",
 "unicode-ident",
]

[[package]]
name = "unicode-ident"
version = "1.0.6"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "84a22b9f218b40614adcb3f4ff08b703773ad44fa9423e4e0d346d5db86e4ebc"

[[package]]
name = "unicode-width"
version = "0.1.10"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "c0edd1e5b14653f783770bce4a4dabb4a5108a5370a5f5d8cfe8710c361f6c8b"

[[package]]
name = "upon"
version = "0.5.0"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "b77ce40602cb1a7dfcdd6336f6d8baa2803c898aafbc0d46156b59727f2e7135"
dependencies = [
 "serde",
 "unicode-ident",
 "unicode-width",
]

[[package]]
name = "winapi"
version = "0.3.9"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "5c839a674fcd7a98952e593242ea400abe93992746761e38641405d28b00f419"
dependencies = [
 "winapi-i686-pc-windows-gnu",
 "winapi-x86_64-pc-windows-gnu",
]

[[package]]
name = "winapi-i686-pc-windows-gnu"
version = "0.4.0"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "ac3b87c63620426dd9b991e5ce0329eff545bccbbb34f3be09ff6fb6ab51b7b6"

[[package]]
name = "winapi-x86_64-pc-windows-gnu"
version = "0.4.0"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "712e227841d057c1ee1cd2fb22fa7e5a5461ae8e48fa2ca79ec42cfc1931183f"

[[package]]
name = "yansi"
version = "0.5.1"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "09041cd90cf85f7f8b2df60c646f853b7f535ce68f85244eb6731cf89fa498ec"
```

**Comment by rossmacarthur** [comment]
Thanks for the report, I guess I've only ever used `goldie` in projects that pull in `serde` with the `derive` feature already and didn't pick this up 🤦‍♂️.

Fixed in 03e3ce6931890bdfb188ff3a9cd3ded4b3e5c645 and released in version 0.4.3

**Comment by ivan-gusiev** [comment]
Omg that was fast! Thanks Ross!
